pkg/test/step_integration_test.go

```diff
@@ -430,3 +430,100 @@ func TestStepFailure(t *testing.T) {
 	errs := step.Run(namespace)
 	assert.Equal(t, len(errs), 1)
 }
+
+func TestAssertCommandsValidCommandRunsOk(t *testing.T) {
+	
+	step := &Step{
+		Name:  			"commandRun",
+		Index: 			0,
+		Logger:          testutils.NewTestLogger(t, ""),
+		Client:          func(bool) (client.Client, error) { return testenv.Client, nil },
+		DiscoveryClient: func() (discovery.DiscoveryInterface, error) { return testenv.DiscoveryClient, nil },
+	}
+
+	// Load test that has an echo command, so it should run ok, and don't return any errors
+	err := step.LoadYAML("step_integration_test_data/assert_commands/valid_command/00-assert.yaml")
+	assert.NoError(t, err)
+
+	errors := step.Run("irrelevant")
+	assert.Equal(t, len(errors), 0)
+}
+
+func TestAssertCommandsMultipleCommandRunsOk(t *testing.T) {
+	
+	step := &Step{
+		Name:  			"commandRun",
+		Index: 			0,
+		Logger:          testutils.NewTestLogger(t, ""),
+		Client:          func(bool) (client.Client, error) { return testenv.Client, nil },
+		DiscoveryClient: func() (discovery.DiscoveryInterface, error) { return testenv.DiscoveryClient, nil },
+	}
+
+	// Load test that has an echo command, so it should run ok, and don't return any errors
+	err := step.LoadYAML("step_integration_test_data/assert_commands/multiple_commands/00-assert.yaml")
+	assert.NoError(t, err)
+
+	errors := step.Run("irrelevant")
+	assert.Equal(t, len(errors), 0)
+}
+
+func TestAssertCommandsMissingCommandFails(t *testing.T) {
+	
+	step := &Step{
+		Name:  			"commandRun",
+		Index: 			0,
+		Logger:          testutils.NewTestLogger(t, ""),
+		Client:          func(bool) (client.Client, error) { return testenv.Client, nil },
+		DiscoveryClient: func() (discovery.DiscoveryInterface, error) { return testenv.DiscoveryClient, nil },
+	}
+
+	// Load test that has an command that is not present (thiscommanddoesnotexist), so it should return an error
+	err := step.LoadYAML("step_integration_test_data/assert_commands/command_does_not_exist/00-assert.yaml")
+	assert.NoError(t, err)
+
+	errors := step.Run("irrelevant")
+	assert.Equal(t, len(errors), 1)
+}
+
+func TestAssertCommandsFailingCommandFails(t *testing.T) {
+	
+	step := &Step{
+		Name:  			"commandRun",
+		Index: 			0,
+		Logger:          testutils.NewTestLogger(t, ""),
+		Client:          func(bool) (client.Client, error) { return testenv.Client, nil },
+		DiscoveryClient: func() (discovery.DiscoveryInterface, error) { return testenv.DiscoveryClient, nil },
+	}
+
+	// Load test that has an command that is present but will allways fail (false), so we should get back the error.
+	err := step.LoadYAML("step_integration_test_data/assert_commands/failing_comand/00-assert.yaml")
+	assert.NoError(t, err)
+
+	errors := step.Run("irrelevant")
+	assert.Equal(t, len(errors), 1)
+}
+
+func TestAssertCommandsShouldTimeout(t *testing.T) {
+	
+	step := &Step{
+		Name:  			"commandRun",
```

Comment:
our linter must not be picking up the integration tests... but these are not properly formatted

Reply:
@jlpedrosa making the mod temporarily to the make from this PR will help ensure alignments on tests.  https://github.com/kudobuilder/kuttl/pull/274

Reply:
@kensipe  if you could merge #274 I could rebase mine and run the test. If I cherry pick that commit here we would have conflicts no?

Reply:
merged :)

Reply:
rebased, and tests fmtd.